src/par_iter/mod.rs

```diff
@@ -363,6 +365,16 @@ pub trait ParallelIterator: Sized {
         reduce(self, reduce_op)
     }
 
+    /// Takes two iterators and creates a new iterator over both.
+    ///
+    /// As with most parallel iterators, the order in which these are
+    /// combined is completely unspecified.
```

Comment:
I feel like I'd just drop it; It feels like it kind of goes w/o saying? but yeah something like "order in which these will be visited when folding/collecting/etc"